modules/solid_mechanics/include/materials/NewtonIterationSingleVariable.h

```diff
@@ -0,0 +1,214 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef NEWTONITERATIONSINGLEVARIABLE_H
+#define NEWTONITERATIONSINGLEVARIABLE_H
+
+#include "InputParameters.h"
+#include "ConsoleStream.h"
+
+/**
+ * Base class that provides capability for Newton return mapping
+ * iterations on a single variable
+ */
+
+class NewtonIterationSingleVariable;
+
+template <>
+InputParameters validParams<NewtonIterationSingleVariable>();
+
+class NewtonIterationSingleVariable
+{
+public:
+  NewtonIterationSingleVariable(const InputParameters & parameters);
+  virtual ~NewtonIterationSingleVariable() {}
+
+protected:
+  /**
+   * Perform the return mapping iterations
+   * @param effective_trial_stress Effective trial stress
+   * @param scalar                 Inelastic strain increment magnitude being solved for
+   * @param console                Console output
+   */
+  void returnMappingSolve(const Real effective_trial_stress,
+                          Real & scalar,
+                          const ConsoleStream & console);
+
+  /**
+   * Compute the maximum permissible value of the scalar.  For some models, the magnitude
+   * of this may be known.
+   * @param effective_trial_stress Effective trial stress
```

Comment:
This is exactly why I don't think the Doxygen comments are particularly useful, but we're supposed to document every single parameter, even if they have self-explanatory names.